### ISSUE TITLE
Current Position as Cover state

### ISSUE THREAD

**Issue by Nigggo** [resoai/TileBoard#164]
Hi there,
I'm just making my first baby steps with TileBoard (and Java Script) and am stuggeling with the covers card.
I'd also +1 for the possibility to show a slider after long press but as an intermediate solution I wanted to show the position as a state instead of just "open" and "close". For some reason the states option does not accept formulas or templates. In the subtitle I could get it to work but I'd prefer to have it up with the state.
Is there a way to solve this or does it require code changes from your side?

```
{
		position: [1, 0],
		type: TYPES.COVER,
		title: 'Esszimmer',
		subtitle: '@attributes.current_position % open',
		id: 'cover.esszimmer',
                 states: {
                       open: 'Open',
                       closed: 'Closed'
                 }
}
```
![image](https://user-images.githubusercontent.com/1877266/46923318-69577f80-d016-11e8-85e7-d3ad7579ca5c.png)

**Comment by resoai** [comment]
Please use state: function

**Comment by Nigggo** [comment]
Thank you. I was 100% sure I tried that before but I must have done something wrong. Now it works :)
What's quite a pitty though is that the tile doesn't seem to fetch the opening position which results in the arrows being incorrectly deactivated if I stop the covers half way (this changes the state to 'open'.
Is there any plan to add full support for covers with position detection?
![image](https://user-images.githubusercontent.com/1877266/46970527-a6c11900-d0b9-11e8-8dd7-9fdeb4b84b1e.png)
![image](https://user-images.githubusercontent.com/1877266/46970679-12a38180-d0ba-11e8-8c59-fcaf394fd7d9.png)

**Comment by resoai** [comment]
We are currently showing arrow controls based on the state (open/closed). Unfortunately I don't have a cover to test it hence the imperfection. Would you be able to show me state dumps at various stages i.e. fully open, fully closed and open half way?

**Comment by Nigggo** [comment]
Of course I will: the state is 'open' for all positions except completely closed as these three screenshots show:
![image](https://user-images.githubusercontent.com/1877266/47039981-78107480-d185-11e8-88b0-fd0f16d6838e.png)

Couldn't you carry over the slider function from the lights card (long press) without much efford? The service to call would be 'cover.set_cover_position' instead of brightness. Or is my lack of coding skills making it look too easy?

**Comment by Krahen** [comment]
Hi, the issue of showing percentage on the state, i've resolved like this:

{
                     position: [0, 0],
                     width: 1,
                     type: TYPES.COVER,
                     id: 'cover.persiana1_level',
                     state: function (item, entity) {
                     	if (this.states['cover.persiana1_level'].attributes.current_position==0)
                     	  return 'Cerrada'
                     	else
					      return 'Abierta '
					         + this.states['cover.persiana1_level'].attributes.current_position + '%';
					   }
                }

Anyway, still there is a problem with the states in this item.. when cover is open but not at 100% it disbled de up button and this  this not should be done until cover is totally up.

thanks.

**Comment by domibarton** [comment]
@Nigggo @Krahen I fixed the cover tile for covers with positional details. Feel free to test it by using [this branch](https://github.com/confirm/TileBoard/tree/cover-partial-open-fix) or manually applying [these changes](https://github.com/confirm/TileBoard/commit/697ed07fc3a331f8c0024bd5523eac3d74a67ab5).

@resoai I'll open a PR, feel free to merge if you're fine with it. I added the issue number to the commit message as you did in your commits and I've added a some notes to it as well. PR coming…
